packages/react-components/src/TxButton.tsx

```diff
@@ -2,67 +2,42 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import { SubmittableExtrinsic } from '@polkadot/api/promise/types';
-import { QueueTx, QueueTxExtrinsicAdd } from './Status/types';
 import { TxButtonProps as Props } from './types';
 
-import React from 'react';
+import React, { useContext } from 'react';
 import { SubmittableResult } from '@polkadot/api';
-import { withApi } from '@polkadot/react-api/hoc';
+import { useApi, useToggle } from '@polkadot/react-hooks';
 import { assert, isFunction, isUndefined } from '@polkadot/util';
 
 import Button from './Button';
-import { QueueConsumer } from './Status/Context';
+import { StatusContext } from './Status';
+import { useTranslation } from './translate';
 
-interface InjectedProps {
-  queueExtrinsic: QueueTxExtrinsicAdd;
-  txqueue: QueueTx[];
-}
+export default function TxButton (props: Props): React.ReactElement<Props> {
```

Comment:
This component alone is probably better suited as a small stand-along PR - just so we can test against existing. (So this one is nicely self-contained)